components/integrations/bots/bots.tsx

```diff
@@ -169,6 +187,7 @@ export default class Bots extends React.PureComponent<Props, State> {
                 accessTokens={(this.props.accessTokens && this.props.accessTokens[bot.user_id]) || {}}
                 actions={this.props.actions}
                 team={this.props.team}
+                fromApp={this.props.appsBotIDs.includes(bot.user_id)}
```

Comment:
Is there an async operation that could momentarily have `fromApp={false}` before the `getAppBotIDs()` response comes back?

Reply:
If getAppBotIDs takes too long, the user will see the whole information, and then disappear when that information finally arrives. We could keep the loading state until we have all the information. What do you think?

Reply:
I think we should keep showing the loading animation. But if the bot id fetch fails, then show all of the information.